### PR TITLE
Cache

### PR DESCRIPTION
**Pull request by kzalys**
Implemented the caching mechanism. #8 
The cache has 1 model so far - User. It is mostly for testing purposes and does not need to be made use of by the user system. The user data is very dynamic and we should only worry about caching it if we have enough time at the end of the project.

### CODE REVIEW

**Review by VladM1998 — COMMENTED** [review]
Everything seems alright ! Waiting for at least another person before we merge it !

**Review by VladM1998 — APPROVED** [review]

**Review by AlexandruTentes — APPROVED** [review]

**Pull request merged**
